neverbleed.c

```diff
@@ -127,10 +136,20 @@ struct st_neverbleed_rsa_exdata_t {
 };
 
 struct st_neverbleed_thread_data_t {
+    int in_flight;
     pid_t self_pid;
     int fd;
 };
 
+void thdata_reset_in_flight(struct st_neverbleed_thread_data_t **_thdata)
+{
+    struct st_neverbleed_thread_data_t *thdata = *_thdata;
+    assert(thdata->in_flight);
+    thdata->in_flight = 0;
+}
+
+#define THDATA_IN_FLIGHT __attribute__ ((__cleanup__(thdata_reset_in_flight))) struct st_neverbleed_thread_data_t
+
```

Comment:
Am I correct to understand that the changes around `struct st_neverbleed_thtread_data_t` are to be removed before merge?

I do not think we'd want to rely on a GCC extension (`__cleanup__`) and the addition of `in_flight` attribute is not required.

Reply:
yes it can be removed, doing this was helpful in finding where there would be transactions being sent while there was one in-flight